docs/api/python/getting-started.md

```diff
@@ -207,324 +369,882 @@ if __name__ == "__main__":
+
+    # Byte-format private key
+    PRIVATE_KEY = bytes.fromhex(os.environ.get("PRIVATE_KEY"))
+
+    # Connect to zkSync network
+    zk_web3 = ZkSyncBuilder.build(PROVIDER)
+
+    # Get account object by providing private key of the sender
+    account: LocalAccount = Account.from_key(PRIVATE_KEY)
+
+    # Provide a compiled JSON source contract
+    contract_path = Path("solidity/incrementer/build/combined.json")
+
+    # Set constructor arguments
+    constructor_arguments = {"_incrementer": 2}
+
+    # Perform contract deployment
+    contract_address = deploy_contract(zk_web3, account, contract_path, constructor_arguments)
+
+    # alternative way specifying constructor arguments using args
+    # constructor_arguments = tuple([2])
+    # change 55 line of code with following:
+    # encoded_constructor = contract_encoder.encode_constructor(*constructor_args)
+
+    execute(zk_web3, account, contract_path, contract_address)
+
+```
+
+### Deploy a contract with dependency
+
+```python
+import os
+from pathlib import Path
+
+from eth_account import Account
+from eth_account.signers.local import LocalAccount
+from web3 import Web3
+
+from zksync2.core.types import EthBlockParams
+from zksync2.manage_contracts.contract_encoder_base import ContractEncoder
+from zksync2.manage_contracts.nonce_holder import NonceHolder
+from zksync2.manage_contracts.precompute_contract_deployer import PrecomputeContractDeployer
+from zksync2.module.module_builder import ZkSyncBuilder
+from zksync2.signer.eth_signer import PrivateKeyEthSigner
+from zksync2.transaction.transaction_builders import TxCreateContract
+
+
+def generate_random_salt() -> bytes:
+    return os.urandom(32)
+
+
+def deploy_contract(
+        zk_web3: Web3, account: LocalAccount, compiled_contract: Path
+):
+    """Deploy compiled contract with dependency on zkSync network using create() opcode
+
+    :param zk_web3:
+        Instance of ZkSyncBuilder that interacts with zkSync network
+
+    :param account:
+        From which account the deployment contract tx will be made
+
+    :param compiled_contract:
+        Compiled contract source.
+
+    :return:
+        Address of deployed contract.
+    """
+    # Get chain id of zkSync network
+    chain_id = zk_web3.zksync.chain_id
+
+    # Signer is used to generate signature of provided transaction
     signer = PrivateKeyEthSigner(account, chain_id)
 
-    counter_contract_bin = read_hex_binary("counter_contract.hex")
+    # Get nonce of ETH address on zkSync network
+    nonce = zk_web3.zksync.get_transaction_count(
+        account.address, EthBlockParams.PENDING.value
+    )
 
-    nonce = zksync_web3.zksync.get_transaction_count(account.address, EthBlockParams.PENDING.value)
-    gas_price = zksync_web3.zksync.gas_price
+    # Deployment of same smart contract (same bytecode) without salt cannot be done twice
+    # Remove salt if you want to deploy contract only once
+    random_salt = generate_random_salt()
```

Comment:
AFAIK 'salt' is only used in 'create2'. In 'create', we should be generating the address based on the account's deployment nonce, that would automatically change.

Reply:
You are right. It's copy paste error.

Reply:
Salt is removed from examples that use create method to deploy contracts